### PR TITLE
Some Spider Changes

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR makes some changes to spiders to make them more balanced, along with some other nice changes to make some subtypes more intuitive and others more fun.  Also contains a somehow unrelated change by allowing player-controlled Lobstrosities to charge after all this time (felt like it'd be a waste of time to PR seperately as well, since it hinges on a small change made to charges in this PR).

Charge cooldown for hostile simplemobs now starts during the initial startup and not when the charge actually occurs.  This is just to prevent players from spamming charge while their first charge is getting prepared.

**BROODMOTHERS**
- Egg laying time increased from 10 seconds to 15 seconds, now uses a variable on the mob itself to determine length.
- Now has a poisonous web projectile which injects 5 toxin into the carbon and a slowdown debuff to any mob when hit

**NURSES**
- Now have a web projectile which deals no damage but slows down targets for 10 seconds

**TARANTULAS**
- Charging button removed
- Charging can now be performed via right-click

**LOBSTROSITIES**
- Can now charge while player-controlled using right-click

## Why It's Good For The Game

There was a request to nerf broodmother egg laying times on the fourms, so here it is.  The web projectiles were originally an idea I had during the era of the spider rework but never added, since it seems broodmothers could use a little help defending themselves starting off and nurses could be more fun to play, I added the projectiles to both of them to make them a little more special.  Finally, with Combat Mode, there's no reason to be using a shoddy UI button for charges when right-click is now completely open and available.  The code to let the player charge is now much simpler, so it felt like a waste if I didn't just give it to lobstrosities as well since its such an easy change to make.

## Changelog
:cl:
add: Broodmother and Nurse spiders now have web projectiles
add: Player-controlled lobstrosities now can charge via right-click
del: Removed Tarantula's charge ability button
qol: Tarantula charge is now mapped to right-click
balance: Broodmother egg-laying time nerfed from 10 seconds to 15 seconds
imageadd: Web projectile
refactor: Re-did tarantula's charge ability to be right-click-based, allowing for the code surrounding it to be greatly reduced and simplified
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
why not give the nurse floor and air blocking webs ? getting  shitmos'd as spider is nasty

**Comment by Nexofas** [comment]
> why not give the nurse floor and air blocking webs ? getting shitmos'd as spider is nasty

Dont know if air blocking webs would solve it but spiders do have a problem against atmos. Making them space-proof while maintaining 02 requirement is what I would go with it.

**Review by Twaticus — APPROVED** [review]

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -228,6 +223,14 @@
 		add_movespeed_modifier(/datum/movespeed_modifier/tarantula_web)
 		silk_walking = FALSE
 
+/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/RangedAttack(atom/target, modifiers)
+	. = ..()
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		if(COOLDOWN_FINISHED(src, charge_cooldown))
+			INVOKE_ASYNC(src, /mob/living/simple_animal/hostile/.proc/enter_charge, target)
+		else
+			to_chat(src, "<span class='notice'>Your charge is still on cooldown!</B></span>")
```
Comment (bobbahbrown) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:232]: hanging ``</B>`` tag -- determine if you meant for this to be bolded or not.

**Comment by carshalash** [comment]
So uh, why the no cooldown on both of the webs pits?

**Comment by RuneQuester** [comment]
The "nerf" is completely meaningless and wont affect spiders *at all*. It just means like 2 less spiders. And them having a ranged attack is utter cancer

**Comment by MrMelbert** [comment]
i'd prefer a bigger nerf tbh, spider wave tactics is still absurdly strong

**Comment by IndieanaJones** [comment]
> So uh, why the no cooldown on both of the webs pits?

Do you know how inheritance works?



> The "nerf" is completely meaningless and wont affect spiders _at all_. It just means like 2 less spiders. And them having a ranged attack is utter cancer. Their main drawback IS the fact they have no ranged attack and rely on ambushes. This is good. Them having a ranged attack with apparently NO COOLDOWN,negates one of their only weaknesses.
> 
> Their main problem,is the infinite respawns they have,for no cost. 15 seconds per egg is still a shit ton of spiders

I'm not putting a cost on basic spider eggs.



> i'd prefer a bigger nerf tbh, spider wave tactics is still absurdly strong

Use stun weapons.

**Comment by IndieanaJones** [comment]
Anyway, I think this is a fine enough nerf for now, since it'll cut down on the reserve amount of eggs the spiders have by 1/3.  We'll see how it goes (no reason to immediately try and butcher them, we're sensible people) and if additional changes are required they'll be made :)

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 1 above)
Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm:232]: What Bee said.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -611,6 +561,34 @@
 		to_chat(ghost, "[link] [my_message]")
 	usr.log_talk(message, LOG_SAY, tag="spider command")
 
+/**
+ * # Web shot
+ *
+ * A projectile shot by spiders, doesn't deal damage but instead grants debuffs to those hit
+ */
+/obj/projectile/spider
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:569]: Rename this to `/obj/projectile/webspit` or something--this implies the projectile is the spider. Spooky.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 3]
```diff
@@ -611,6 +561,34 @@
...
+	name = "web spit"
+	nodamage = TRUE
+	damage = 0
+	icon_state = "web_spit"
+
+/obj/projectile/spider/on_hit(atom/target, blocked = FALSE)
+	if(isliving(target) && blocked < 100)
+		var/mob/living/l_target = target
+		l_target?.apply_status_effect(STATUS_EFFECT_WEBBED)
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:578]:
```suggestion
		var/mob/living/living_target = target
		living_target?.apply_status_effect(STATUS_EFFECT_WEBBED)
```

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/lobstrosity.dm [hunk 4]
```diff
@@ -34,6 +34,14 @@
 	footstep_type = FOOTSTEP_MOB_CLAW
 	gold_core_spawnable = HOSTILE_SPAWN
 
+/mob/living/simple_animal/hostile/asteroid/lobstrosity/RangedAttack(atom/target, modifiers)
+	. = ..()
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/lobstrosity.dm:38]: Are you sure RangedAttack should be calling `..()` if its not a right click? Sounds like a recipe for "Oops, right clicking does both attacks now" if another one is added.

Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/lobstrosity.dm:38]: To be honest, I think this should just be moved out to a `proc/ranged_attack_secondary` like everything else is, maybe reusing the existing _SECONDARY defines as well.

Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/mining_mobs/lobstrosity.dm:38]: This same feedback goes for the tarantula right-click as well.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by RuneQuester** [comment]
A reasonable buff to spiders would be fixing/buffing the webs,as humans are only slowed on the initial movement into webs,and suffer no movement penalty after,which negates the whole purpose of webs

**Comment by IndieanaJones** [comment]
> A reasonable buff to spiders would be fixing/buffing the webs,as humans are only slowed on the initial movement into webs,and suffer no movement penalty after,which negates the whole purpose of webs. I
> 
> In addition,making spiders inject venom instead of toxin would be nice

That's incorrect, webs have a chance to stall any mob from moving onto its position when it attempts on a random chance basis.  I'm personally not a huge fan of RNG, but the whole point is to simply encourage people to not walk over webs, which I believe it does successfully.  

Also, buffing the spider's toxins into venom is probably not a good idea.  I'd go as far to say they're my least favorite aspect of spiders (unseen damage over time isn't exactly engaging for either player), but that's just me.

**Pull request closed without merging**